myaccount.php

```diff
@@ -66,62 +68,76 @@
-				if ($order !== false) {
-					// Check if player and account matches
-					if ($session_user_id == $player['account_id'] && $session_user_id == $order['account_id']) {
-						// Check if new name is not occupied
-						$exist = mysql_select_single("SELECT `id` FROM `players` WHERE `name`='$newname';");
-						if (!$exist) {
-							// Check if new name follow rules
-							$newname = validate_name($newname);
-							if ($newname !== false) {
-								$error = false;
-								// name restriction
-								$resname = explode(" ", $_POST['name']);
-								foreach($resname as $res) {
-									if(in_array(strtolower($res), $config['invalidNameTags'])) {
-										$error = true;
-									}
-									else if(strlen($res) == 1) {
-										$error = true;
-									}
-								}
-								// Check name for illegal characters.
-								function checkNewNameForIllegal($name) {
-									if (preg_match('#^[\0-9åäö&()+%/*$€é,.\'"-]*$#i', $name)) {
-										return true;
-									}
-									return false;
-								}
-								if (checkNewNameForIllegal($newname)) {
-									$error = true;
-									echo 'This name contains illegal characters.';
-								}
-								if ($error === false) {
-									// Change the name!
-									mysql_update("UPDATE `players` SET `name`='$newname' WHERE `id`='".$player['id']."' LIMIT 1;");
-									mysql_delete("DELETE FROM `znote_shop_orders` WHERE `id`='".$order['id']."' LIMIT 1;");
-								}
-							} else echo 'Name validation failed, use another name.';
-						} else echo 'The character name you wish to change to already exist.';
-					} else echo 'Failed to sync your account. :|';
-				} else echo 'Did not find any name change tickets, buy them in our <a href="shop.php">shop!</a>';
+				$accountId = $player['account_id'];
+				$order = mysql_select_single("SELECT `id`, `account_id` FROM `znote_shop_orders` WHERE `type`='4' AND `account_id` = '$accountId' LIMIT 1;");
+				if ($order === false) {
+					$errors[] = 'Did not find any name change tickets, buy them in our <a href="shop.php">shop!</a>';
+				}
+
+				// Check if player and account matches
+				if ($session_user_id != $accountId || $session_user_id != $order['account_id']) {
+					$errors[] = 'Failed to sync your account. :|';
+				}
+
+				$newname = validate_name($newname);
+				if ($newname === false) {
+					$errors[] = 'Your name can not contain more than 2 words.';
+				} else {
+					if (empty($newname)) {
+						$errors[] = 'Please enter a name!';
+					} else if (user_character_exist($newname) !== false) {
+						$errors[] = 'Sorry, that character name already exist.';
+					} else if (!preg_match("/^[a-zA-Z_ ]+$/", $newname)) {
+						$errors[] = 'Your name may only contain a-z, A-Z and spaces.';
+					} else if (strlen($newname) < $config['minL'] || strlen($newname) > $config['maxL']) {
+						$errors[] = 'Your character name must be between 4 - 20 characters long.';
```

Comment:
make it uses the values from variables minL and maxL instead of 4 - 20 so if anyone edit the config value it gets updated here as well